pool/pool.go

```diff
@@ -208,7 +209,11 @@ func (p *Pool) StoreTx(ctx context.Context, tx types.Transaction, ip string, isW
 			log.Errorf("error adding event: %v", err)
 		}
 		// Do not add tx to the pool
-		return ErrOutOfCounters
+		err = ErrOutOfCounters
+		if preExecutionResponse.OOCError != nil {
+			err = fmt.Errorf("failed to add tx to the pool: %v", preExecutionResponse.OOCError.Error())
+		}
+		return err
```

Comment:
How is it possible to have a ErrOutOfCounters if the preExecutionResponse.OOCError is nil?? I mean if the executor didn't return the OOCError it shouldn't be possible to have this situation

Reply:
I understand OOC is the default case and then it gets overwritten if there is an OOC

Reply:
> How is it possible to have a ErrOutOfCounters if the preExecutionResponse.OOCError is nil?? I mean if the executor didn't return the OOCError it shouldn't be possible to have this situation

yes, I just added this check just in case a code in the future would change the logic, to keep this code protected, just in case a new OOC code is added, we change part of the code and forget to change some other part, this could break the pool.

But I'll be refactoring the code to avoid this double check.